ccxt.js

```diff
@@ -13161,26 +13193,25 @@ var poloniex = {
     },
 
     async fetchMyTrades (market = undefined, params = {}) {
-        let now = this.seconds ();
+        let m = undefined;
+        if (market)
+            m = this.market (market);
+        let pair = m ? m['id'] : 'all';
         let request = this.extend ({
-            'currencyPair': 'all',
+            'currencyPair': pair,
             'end': this.seconds (), // last 50000 trades by default
         }, params);
-        if (market) {
-            let m = this.market (market);
-            request['currencyPair'] = m['id'];
-        }
         let trades = await this.privatePostReturnTradeHistory (request);
-        if (market)
+        if (market) // This is a hard to read control flow
```

Comment:
I completely missed the return when I first read this code, so maybe it's a hint that this flow is somewhat misleading or I'm dyslexic )